pkg/lib/cockpit-components-context-menu.tsx

```diff
@@ -27,33 +26,36 @@ import "context-menu.scss";
 /*
  * A context menu component
  *
- * It requires two properties:
- *  - parentId, area in which it listens to left button click
- *  - children, a MenuList to be rendered in the context menu
+ * It has two properties:
+ *  - parentId (required), area in which it listens to left button click
+ *  - children (optional), a MenuList to be rendered in the context menu
  */
-export const ContextMenu = ({ parentId, children }) => {
+export const ContextMenu = ({ parentId, children } : {
+    parentId: string,
+    children?: React.ReactNode[],
+}) => {
     const [visible, setVisible] = React.useState(false);
-    const [event, setEvent] = React.useState(null);
-    const root = React.useRef(null);
+    const [event, setEvent] = React.useState<MouseEvent | null>(null);
+    const root = React.useRef<HTMLDivElement>(null);
```

Comment:
Curious -- how can that get away without `HTMLDivElement | null`?

Reply:
There's some magic here where the resulting type is the union of the given type and the literal type of the thing passed in as the initial value.